### PR TITLE
splunk-s3-backup-play

### PR DESCRIPTION
**Pull request by nadeemshahzad**
@edx/devops Kindly review

### CODE REVIEW

File: playbooks/edx-east/splunk_config_backup.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  hosts: all
+  vars:
+    splunk_config_dir: "/opt/splunk/etc"
+    splunk_config_archive: "{{ date }}.tar.gz"
+    splunk_s3_backup_tempdir: "/var/tmp"
+    splunk_s3_bucket: "{{ SPLUNK_S3_BACKUPS_BUCKET }}"
+    date: "{{ lookup('pipe', 'date +%Y-%m-%d') }}"
+  tasks:
+    - name: archive splunk configuration dir
+      command: >
```
Comment (jibsheet) [inline playbooks/edx-east/splunk_config_backup.yml]: Is the > just because of the line length?

File: playbooks/edx-east/splunk_config_backup.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+- name: Backup splunk configurations
+  hosts: all
+  vars:
+    splunk_config_dir: "/opt/splunk/etc"
+    splunk_config_archive: "{{ date }}.tar.gz"
+    splunk_s3_backup_tempdir: "/var/tmp"
+    splunk_s3_bucket: "{{ SPLUNK_S3_BACKUPS_BUCKET }}"
+    date: "{{ lookup('pipe', 'date +%Y-%m-%d') }}"
```
Comment (jibsheet) [inline playbooks/edx-east/splunk_config_backup.yml]: Can we include more of the timestamp (minutes/hour should suffice)

Just thinking I would totally run this play to back up before making a config change, and that would overwrite the backup from earlier in the day.

That way I could have splunk-backup.2017-04-02T2048.tar.gz

File: playbooks/edx-east/splunk_config_backup.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+- name: Backup splunk configurations
+  hosts: all
+  vars:
+    splunk_config_dir: "/opt/splunk/etc"
+    splunk_config_archive: "{{ date }}.tar.gz"
```
Comment (jibsheet) [inline playbooks/edx-east/splunk_config_backup.yml]: Can you consume a prefix of some kind?  we run several splunk servers and should back all of them up (I know the ticket is concentrated on one in particular)

Comment (nadeemshahzad) [reply playbooks/edx-east/splunk_config_backup.yml]: @jibsheet Added prefix with splunk server host id

File: playbooks/edx-east/splunk_config_backup.yml (hunk 2 above)
Comment (nadeemshahzad) [reply playbooks/edx-east/splunk_config_backup.yml]: @jibsheet Include (hour/minutes) in timestamp and make date persistent.

File: playbooks/edx-east/splunk_config_backup.yml (hunk 1 above)
Comment (nadeemshahzad) [reply playbooks/edx-east/splunk_config_backup.yml]: @jibsheet It was for line length , but it is ok to remove >

**Review by jibsheet — APPROVED** [review]
Changes look good - I'll want to see how you pass in the args in the Jenkins DSL, but I think this is good.

**Pull request merged**